src/choleskyQR.jl

```diff
@@ -1,56 +1,77 @@
+"""
+Represent an Orthogonal polynomial which has a conversion operator to P, that is, Q = P * R.
+"""
+struct ConvertedOrthogonalPolynomial{T, WW<:AbstractQuasiVector{T}, XX, RR, PP} <: OrthogonalPolynomial{T}
+    weight::WW
+    X::XX # jacobimatrix
+    R::RR # conversion to P
+    P::PP
+end
+
+# transform to P * U if needed for differentiation, etc.
+arguments(::ApplyLayout{typeof(*)}, Q::ConvertedOrthogonalPolynomial) = Q.P, Q.R
+
+# also change all the NormalizedOPLayout
+@inline copy(L::Ldiv{Lay,<: AbstractConvertedOPLayout}) where Lay = copy(Ldiv{Lay,ApplyLayout{typeof(*)}}(L.A, L.B))
+
+OrthogonalPolynomial(w::AbstractQuasiVector) = OrthogonalPolynomial(w, orthonormalpolynomial(singularities(w)))
+function OrthogonalPolynomial(w::AbstractQuasiVector, P::AbstractQuasiMatrix)
+    X = cholesky_jacobimatrix(w, P)
+    ConvertedOrthogonalPolynomial(w, X, X.dv.U, P)
+end
+
+
+
 """
 cholesky_jacobimatrix(w, P)
 
 returns the Jacobi matrix `X` associated to a quasi-matrix of polynomials
 orthogonal with respect to `w(x) w_p(x)` where `w_p(x)` is the weight of the polynomials in `P`.
 
 The resulting polynomials are orthonormal on the same domain as `P`. The supplied `P` must be normalized. Accepted inputs are `w` as a function or `W` as an infinite matrix representing multiplication with the function `w` on the basis `P`.
-
-An optional bool can be supplied, i.e. `cholesky_jacobimatrix(sqrtw, P, false)` to disable checks of symmetry for the weight multiplication matrix and orthonormality for the basis (use with caution).
 """
-function cholesky_jacobimatrix(w::Function, P::OrthogonalPolynomial, checks::Bool = true)
-    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
-    W = Symmetric(P \ (w.(axes(P,1)) .* P)) # Compute weight multiplication via Clenshaw
-    return cholesky_jacobimatrix(W, P, false)     # At this point checks already passed or were entered as false, no need to recheck
+cholesky_jacobimatrix(w::Function, P) = cholesky_jacobimatrix(w.(axes(P,1)), P)
+
+function cholesky_jacobimatrix(w::AbstractQuasiVector, P)
+    Q = normalized(P)
+    W = Symmetric(Q \ (w .* Q)) # Compute weight multiplication via Clenshaw
+    return cholesky_jacobimatrix(W, Q)
 end
-function cholesky_jacobimatrix(W::AbstractMatrix, P::OrthogonalPolynomial, checks::Bool = true)
-    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
-    checks && !(W isa Symmetric) && error("Weight modification matrix must be symmetric.")
-    return SymTridiagonal(CholeskyJacobiBands{:dv}(W,P),CholeskyJacobiBands{:ev}(W,P))
+function cholesky_jacobimatrix(W::AbstractMatrix, Q)
+    isnormalized(Q) || error("Polynomials must be orthonormal")
+    issymmetric(W) || error("Weight modification matrix must be symmetric.")
```

Comment:
The reason I checked via `isa Symmetric` is that `issymmetric()` tended to end up in infinite loops for certain types for me (maybe it was just SymTridiagonal though which would return false here anyway). Your inclusion of it here suggests it works for you? To give an example where it gets stuck for me:

```julia
julia> T = Normalized(Jacobi(0, 1))
Normalized(Jacobi(0.0, 1.0))

julia> X = jacobimatrix(T)
ℵ₀×ℵ₀ LazyBandedMatrices.SymTridiagonal{Float64, ApplyArray{Float64, 1, typeof(vcat), Tuple{Float64, BroadcastVector{Float64, typeof(/), Tuple{Float64, BroadcastVector{Float64, typeof(*), Tuple{InfiniteArrays.InfStepRange{Float64, Float64}, InfiniteArrays.InfStepRange{Float64, Float64}}}}}}}, BroadcastVector{Float64, typeof(sqrt), Tuple{BroadcastVector{Float64, typeof(*), Tuple{BroadcastVector{Float64, typeof(/), Tuple{BroadcastVector{Float64, typeof(*), Tuple{InfiniteArrays.InfStepRange{Float64, Float64}, InfiniteArrays.InfStepRange{Float64, Float64}}}, BroadcastVector{Float64, typeof(*), Tuple{InfiniteArrays.InfStepRange{Float64, Float64}, InfiniteArrays.InfStepRange{Float64, Float64}}}}}, ApplyArray{Float64, 1, typeof(vcat), Tuple{Float64, BroadcastVector{Float64, typeof(/), Tuple{BroadcastVector{Float64, typeof(*), Tuple{InfiniteArrays.InfStepRange{Int64, Int64}, InfiniteArrays.InfStepRange{Float64, Float64}}}, BroadcastVector{Float64, typeof(*), Tuple{InfiniteArrays.InfStepRange{Float64, Float64}, InfiniteArrays.InfStepRange{Float64, Float64}}}}}}}}}}}} with indices OneToInf()×OneToInf():
 0.333333  0.471405    ⋅          ⋅         ⋅         ⋅           ⋅          …  
 0.471405  0.0666667  0.489898    ⋅         ⋅         ⋅           ⋅             
  ⋅        0.489898   0.0285714  0.494872   ⋅         ⋅           ⋅             
  ⋅         ⋅         0.494872   0.015873  0.496904   ⋅           ⋅             
  ⋅         ⋅          ⋅         0.496904  0.010101  0.49793      ⋅             
  ⋅         ⋅          ⋅          ⋅        0.49793   0.00699301  0.498519    …  
  ⋅         ⋅          ⋅          ⋅         ⋅        0.498519    0.00512821     
  ⋅         ⋅          ⋅          ⋅         ⋅         ⋅          0.498888       
  ⋅         ⋅          ⋅          ⋅         ⋅         ⋅           ⋅             
 ⋮                                                   ⋮                       ⋱  

julia> issymmetric(X)
# infinite loop
```

Reply:
Wait, why isn't `X` a `SymTridiagonal`??

Reply:
Actually I decided to just remove the check: checking the type is unreliable and too restrictive, and checking `issymmetric` might end up in an infinite loop.

Reply:
This is what I basically ended up doing by having the bool set to `false`. :smile: I agree that the checking is annoying, if it fails it fails in relatively predictable ways anyway